packages/shared/components/atoms/SideDrawer.svelte

```diff
@@ -0,0 +1,53 @@
+<script lang="ts">
+    import { fade, fly } from 'svelte/transition'
+    import { appSettings } from '@core/app'
+
+    export let onClose: () => unknown = () => {}
+    export let preventClose: boolean = false
+
+    const DRAWER_ANIMATION_DURATION_MS = 200
+    const directon = { x: 100 }
+
+    const position = 0
+
+    function onOverlayClick(): void {
+        if (!preventClose) {
+            onClose && onClose()
+        }
+    }
+</script>
+
+<drawer class="fixed top-0 left-0 w-screen h-screen">
+    <overlay
+        in:fade|local={{ duration: DRAWER_ANIMATION_DURATION_MS }}
+        out:fade|local={{ duration: DRAWER_ANIMATION_DURATION_MS }}
+        on:click={onOverlayClick}
+        on:keydown={() => {}}
+        class="fixed top-0 left-0 w-full h-full z-0 bg-gray-700 dark:bg-gray-900 bg-opacity-60 dark:bg-opacity-60"
+    />
+    <panel
+        in:fly|local={{ ...directon, duration: DRAWER_ANIMATION_DURATION_MS }}
+        out:fly|local={{ ...directon, duration: DRAWER_ANIMATION_DURATION_MS }}
+        class="h-screen"
+        class:darkmode={$appSettings.darkMode}
+        style:--right={`${position}px`}
+    >
+        <slot />
+    </panel>
+</drawer>
+
+<style type="text/scss">
+    panel {
+        @apply fixed w-full max-h-full;
+        @apply flex flex-col flex-auto overflow-hidden;
+        @apply pt-7 px-5;
```

Comment:
With this padding the title text is slightly closer to the left of the side drawer, making it look just slightly uneven. The padding from all sides (possibly not the bottom) should be the same. 

Some side drawers will aslo have a back button to the left of the title text, so it would be good to see what that looks like here (sorry if that wasn't a task requirement).